superset/assets/javascripts/explorev2/components/controls/VizTypeControl.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Label, FormControl, Modal } from 'react-bootstrap';
+import visTypes from '../../stores/visTypes';
+
+const IMG_SIZE = 150;
+
+const propTypes = {
+  description: PropTypes.string,
+  label: PropTypes.string,
+  name: PropTypes.string.isRequired,
+  onChange: PropTypes.func,
+  value: PropTypes.string.isRequired,
+};
+
+const defaultProps = {
+  onChange: () => {},
+};
+
+export default class VizTypeControl extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false,
+      filter: '',
+    };
+    this.toggleModal = this.toggleModal.bind(this);
+    this.changeSearch = this.changeSearch.bind(this);
+  }
+  onChange(vizType) {
+    this.props.onChange(vizType);
+    this.setState({ showModal: false });
+  }
+  toggleModal() {
+    this.setState({ showModal: !this.state.showModal });
+  }
+  changeSearch(event) {
+    this.setState({ filter: event.target.value });
+  }
+  render() {
+    const filter = this.state.filter;
+    return (
+      <div>
+        <Label>{visTypes[this.props.value].label}</Label>
+        <Button onClick={this.toggleModal} bsSize="sm" className="l-m-5">
```

Comment:
could we stick to a similar format for editing controls, and add an edit link rather than the button with viz types?

<img width="890" alt="screenshot 2017-05-23 11 52 28" src="https://cloud.githubusercontent.com/assets/130878/26371325/44cd76ea-3faf-11e7-8990-0696cfa4ef2e.png">